CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+cmake_minimum_required(VERSION 3.24)
+project(llm.c LANGUAGES C)
+# project(llm.c LANGUAGES C CXX CUDA)
+
+# Put binaries and libraries in the same location.
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
+
+# Always export compile_commands.json for lsp like clangd.
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+# We don't support this compiler.
+if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  message(FATAL_ERROR "This compiler is not supported")
+endif()
+
+# Release by default if not specified.
+if (NOT EXISTS ${CMAKE_BINARY_DIR}/CMakeCache.txt)
+  if (NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "" FORCE)
+  endif()
+endif()
+
+# option(PRECISION "Precision settings" BF16)
```

Comment:
What's the default `Precision settings`? If we don't use `BF16`, then what will it be?

Reply:
it's default to `BF16`, but you can also use `FP32` or `FP16`